From cafb674fb230d16b4302ad453bda103fa991914c Mon Sep 17 00:00:00 2001
From: David Reaver <me@davidreaver.com>
Date: Mon, 3 Feb 2025 13:13:09 -0800
Subject: [RFC PATCH 1/7] debugfs: Add temporary "#define debugfs_node dentry"
To: Greg Kroah-Hartman <gregkh@linuxfoundation.org>,
    Rafael J. Wysocki <rafael@kernel.org>,
    Danilo Krummrich <dakr@kernel.org>
Cc: Steven Rostedt <rostedt@goodmis.org>,
    Christian Brauner <brauner@kernel.org>,
    Alexander Viro <viro@zeniv.linux.org.uk>,
    Tejun Heo <tj@kernel.org>,
    linux-fsdevel@vger.kernel.org,
    cocci@inria.fr,
    linux-kernel@vger.kernel.org,
    David Reaver <me@davidreaver.com>

Add a temporary "#define debugfs_node dentry" to facilitate migrating
debugfs APIs from dentry to a new opaque descriptor, debugfs_node.
This change allows subsequent commits to apply a Coccinelle script and
manual fixups before replacing the #define with an actual struct
declaration.

To minimize the number of additional #include statements, the #define is
also placed in dcache.h, right below the definition of struct dentry.

Signed-off-by: David Reaver <me@davidreaver.com>
---
 include/linux/dcache.h  | 2 ++
 include/linux/debugfs.h | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/include/linux/dcache.h b/include/linux/dcache.h
index 4afb60365675..4b0c11cd3d50 100644
--- a/include/linux/dcache.h
+++ b/include/linux/dcache.h
@@ -127,6 +127,8 @@ struct dentry {
 	} d_u;
 };
 
+#define debugfs_node dentry
+
 /*
  * dentry->d_lock spinlock nesting subclasses:
  *
diff --git a/include/linux/debugfs.h b/include/linux/debugfs.h
index fa2568b4380d..e6ee571e8c36 100644
--- a/include/linux/debugfs.h
+++ b/include/linux/debugfs.h
@@ -21,6 +21,8 @@
 struct device;
 struct file_operations;
 
+#define debugfs_node dentry
+
 struct debugfs_blob_wrapper {
 	void *data;
 	unsigned long size;
